### PR TITLE
add enum type to handle "exit" command

### PR DESCRIPTION
**Pull request by sshang0309**

### CODE REVIEW

**Comment by sshang0309** [comment]
I always use the tab to indent the file and the tab is 4 spaces. But every time I upload it to the github, it becomes not 4 spaces.

**Comment by yichenshen** [comment]
Did you follow the steps here: http://stackoverflow.com/questions/407929/how-do-i-change-eclipse-to-use-spaces-instead-of-tabs?

The tab may be displayed as 4 spaces long, but it's still a tab character on the file if you didn't set it right. You have to make sure when you press the <- button it goes back space by space.

Please reformat before merge.

File: src/lifetracker/logic/CommandLineResult.java [hunk 1]
```diff
@@ -35,4 +40,20 @@ public void addResultLine(String resultLine) {
 
         resultLines.add(resultLine);
     }
+    
+    @Override
+    public void setType(String commandString){
+    	if(commandString.equals("exit")){
+    		this.commandType = CommandType.EXIT;
+    		System.exit(0);
```
Comment (yichenshen) [inline src/lifetracker/logic/CommandLineResult.java]: Do not exit here! 

This is a premature termination. Just pass the signal to the UI and let it handle the exit. If you simply exit here, the storage will shutdown immediately and cause file corruptions.

File: src/lifetracker/logic/CommandLineResult.java [hunk 2]
```diff
@@ -35,4 +40,20 @@ public void addResultLine(String resultLine) {
 
         resultLines.add(resultLine);
     }
+    
+    @Override
+    public void setType(String commandString){
+    	if(commandString.equals("exit")){
```
Comment (yichenshen) [inline src/lifetracker/logic/CommandLineResult.java]: You might want to do this part in the main logic itself... Up to you.

File: src/lifetracker/logic/CommandLineResult.java [hunk 3]
```diff
@@ -4,9 +4,14 @@
 import java.util.List;
 
 public class CommandLineResult implements ExecuteResult {
+	
+    public enum CommandType {
```
Comment (yichenshen) [inline src/lifetracker/logic/CommandLineResult.java]: Define the enum in the interface.

If you put it in the implementation, then the implementation depends on the abstraction, and not the other way round, which defeats the purpose of the abstraction in the first place.

**Pull request merged**